packages/@aws-cdk/core/lib/secret-value.ts

```diff
@@ -52,6 +59,11 @@ export class SecretValue extends Intrinsic {
     ];
 
     const dyref = new CfnDynamicReference(CfnDynamicReferenceService.SECRETS_MANAGER, parts.join(':'));
+
+    if (secretId.startsWith('arn:')) {
+      const secretArn = options.jsonField ? `${secretId}:${options.jsonField}` : secretId;
```

Comment:
1. I find it incredibly confusing that `secretArn` is not actually an ARN, but also contains the qualifiers...
2. Just `jsonField` is not enough - CodeBuild also supports `versionStage` and `versionId` (https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#build-spec.env.secrets-manager).

This all tells me that `secretArn?: string` is not good enough here.

Let's create a new interface for this concept, let's say `SecretQualifier`:

```ts
export interface SecretQualifier {
  readonly secretId: string;
  readonly jsonField?: string;
  readonly versionStage?: string;
  readonly versionId?: string;
}
```

And let's make the new field in `SecretValue` `readonly secretQualifier?: SecretQualifier`.